amity/amity.py

```diff
@@ -0,0 +1,255 @@
+                                              new_person, 
+                                              wants_accommodation,
+                                              rooms
+                                           )
+
+    def randomly_allocate_rooms(self, person, wants_accommodation, rooms):
+        """Allocate person to room
+        Both staff and fellow are assigned office spaces
+        Only fellow who indicates interest is assigned living space
+        Args:
+        person -- the person of interest
+        wants_accommodation -- indicates if person wants accommodation or not
+        rooms -- list of all created rooms
+        Returns:
+        a call to add_persons_to_room method 
+        """
+        allocated_space  = []
+        if self.check_room_type_has_vacancy(rooms, "office"):
+            allocated_space.append(self.randomly_pick_a_room(rooms, "office"))
+        if person.type == "fellow" and wants_accommodation == "Y" \
+            and self.check_room_type_has_vacancy(rooms, "living space"):
+            allocated_space.append(self.randomly_pick_a_room(rooms, 
+                                                             "living space"
+                                                            ))
+        return self.add_person_to_rooms(person, allocated_space)
+        
+    def check_room_type_has_vacancy(self, rooms, room_type):
+        """
+        Args:
+        rooms -- all rooms in database
+        room_type -- the room type of interest
+        Returns:
+        True -- if there is a vacant room of a particular type. 
+        False -- if there is none
+        """
+        for room in rooms:
+            if self.check_that_room_is_vacant(room, room_type):
+                return True
+        print "There is no vacant {0} currently in database!!!". \
+            format(room_type.capitalize())
+        return False
+
+    def add_person_to_rooms(self, person, rooms):
+        """Update Person rooms column with the new rooms assigned
+        Args:
+        person -- the person of interest
+        rooms -- the rooms we want to allocate to person
+        Returns:
+        person -- the updated person object reflecting the allocation
+        """
+        for room in rooms:
+            person.room.append(room)
+            print "{0} {1} was assigned {2} {3}.".format(person.first_name, 
+                person.last_name, room.name.capitalize(), room.type)
+        return person
+
+    def randomly_pick_a_room(self, rooms, room_type):
+        """
+        Randomly pick an office or living space based on supplied argument
+        Args:
+        rooms -- all rooms in database
+        room_type -- the room type of interest
+        Returns:
+        room -- a randomly selected room
+        """
+        last_room_index = len(rooms) - 1
+        random_index = random.randint(0, last_room_index)
+        searching_for_vacant_room = True
+        while searching_for_vacant_room:
```

Comment:
**Code Quality - Nitpick:** Will be good to skip one line here for better readability.